source/Engine.cpp

```diff
@@ -1064,8 +1109,8 @@ void Engine::Draw() const
 	{
 		Point size(highlightSprite->Width(), highlightSprite->Height());
 		const Color &color = *colors.Get("flagship highlight");
-		// The flagship is always in the dead center of the screen.
-		OutlineShader::Draw(highlightSprite, Point(), size, color, highlightUnit, highlightFrame);
+		OutlineShader::Draw(highlightSprite, (highlightLocation - center) * zoom,
```

Comment:
high lightsprite only exists if flagship exists so we dont actually need a variable called highlightLocation and can just use flaghsip->Center here

Reply:
> high lightsprite only exists if flagship exists so we dont actually need a variable called highlightLocation and can just use flaghsip->Center here

The highlight lags forward by a frame if we just use thed flagship's position. There's a reason things like status overlays and alert labels have separate structs that encode the data you need to draw.